tests/rados/test_pool_osd_recovery.py

```diff
@@ -0,0 +1,348 @@
+    ocount_pre = len(get_osd_details(node=cephadm))
+    add_osd(ceph_cluster=ceph_cluster, config=osd_args)
+    if not ocount_pre <= len(get_osd_details(node=cephadm)):
+        log.error("New OSDs were not added into the cluster")
+        raise Exception("Execution error")
+
+    log.info("Deployed new hosts and deployed OSDs on them")
+    # Waiting for recovery to post OSD addition into cluster
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj)
+    log.debug("PG's are active + clean post OSD addition")
+
+    # Checking cluster health after the tests
+    method_should_succeed(rados_obj.run_pool_sanity_check)
+
+    log.info(
+        "Cluster is healthy post addition of New hosts and osds."
+        " Proceeding to remove one of the added host and it's OSDs"
+    )
+
+    # Removing newly added OSD host and checking status
+    rm_host_id = config.get("remove_host", "node13")
+    rm_host = get_node_by_id(ceph_cluster, rm_host_id)
+    log.info(f"Identified host : {rm_host.hostname} to be removed from the cluster")
+
+    # Starting to drain the host
+    drain_cmd = f"ceph orch host drain {rm_host.hostname}"
+    cephadm.shell([drain_cmd])
+    # Sleeping for 2 seconds for removal to have started
+    time.sleep(2)
+    log.debug(f"Started drain operation on node : {rm_host.hostname}")
+
+    status_cmd = "ceph orch osd rm status -f json"
+    end_time = datetime.datetime.now() + datetime.timedelta(seconds=3000)
+    flag = False
+    while end_time > datetime.datetime.now():
+        out, err = cephadm.shell([status_cmd])
+        try:
+            drain_ops = json.loads(out)
+            for entry in drain_ops:
+                log.debug(
+                    f"Drain operations are going on host {rm_host.hostname} \nOperations: {entry}"
+                )
+        except json.JSONDecodeError:
+            log.info(f"Drain operations completed on host : {rm_host.hostname}")
+            flag = True
+            break
+        except Exception as error:
+            log.error(f"Hit issue during drain operations: {error}")
+            raise Exception(error)
+        log.debug("Sleeping for 10 seconds and checking again....")
+        time.sleep(10)
+
+    if not flag:
+        log.error(
+            "Drain operation not completed on the cluster even after 3000 seconds"
+        )
+        raise Exception("Execution Error")
+
+    log.info(
+        f"Completed drain operation on the host. {rm_host.hostname}\n Removing host from the cluster"
+    )
+    time.sleep(5)
+    rm_cmd = f"ceph orch host rm {rm_host.hostname} --force"
+    cephadm.shell([rm_cmd])
+    time.sleep(5)
+
+    # Checking if the host still exists on the cluster
+    ls_cmd = "ceph orch host ls"
+    hosts = rados_obj.run_ceph_command(cmd=ls_cmd)
+    for host in hosts:
+        if host["hostname"] == rm_host.hostname:
+            log.error(f"Host : {rm_host.hostname} still present on the cluster")
+            raise Exception("Host not removed error")
```

Comment:
Small suggestion, instead of traversing complete list of hosts on the cluster we can check out output of cmd: ` ceph orch host ls --host_pattern rm_host.hostname` to determine whether the host still exists

Reply:
In this case, If the node isn't present, the o/p won't be in json, which will be bit more complicated.

If the above comment is a requirement, I will update the code. Let me know.

Reply:
The o/p would be an empty list if a host does not exist, however, this is very trivial and works either way, resolving this thread.